.github/workflows/javascript.yml

```diff
@@ -0,0 +1,98 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2023 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: JavaScript Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  lint:
+    name: Perform linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        with:
+          path: benchexec/tablegenerator/react-table/node_modules
```

Comment:
Thanks for thinking of caching, this is good!

However, as far as I understand [`npm ci`](https://docs.npmjs.com/cli/v9/commands/npm-ci/#description), it always deletes `node_modules`. On GitLab CI we instead cache the cache directory of npm, such that at least the downloading of packages is skipped:
https://github.com/sosy-lab/benchexec/blob/2217e9f2f6b2800b749bd2cb9bd136d17225691a/.gitlab-ci.yml#L15-L28

Can we do the same here? And can we also set `prefer_offline` such that `npm` doesn't attempt to look up stuff online as long as it is not necessary?

Reply:
Got it! I would look into the ways of doing the same. Using `npm ci` is preferred as it would respect the `package-lock.json` and would not fetch the lastest updates.

Reply:
I couldn't find any resource for setting `prefer_offline` setting for this workflow, but I found this [article](https://www.voorhoede.nl/en/blog/super-fast-npm-install-on-github-actions/), which in my opinion does a great job of explaining the two different caching strategies and what the ideal idiomatic think should be. Using `npm ci` along with caching `.npm` seems to our best bet!